src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -724,7 +724,7 @@ private void generateMethod(ClassWriter cw, String className) {
 
             MethodVisitor mv = cw.visitMethod(accessFlags,
                     method.getName(), desc, null,
-                    typeNames(Arrays.asList(exceptionTypes)));
+                    typeNames(List.of(exceptionTypes)));
```

Comment:
Since `exceptionTypes` is an array, it might be better to keep using `Arrays.asList`, or add an overload for `typeNames` that works on class arrays.

Reply:
Usually a method declares either no exception, or a couple of them. In the first case `List.of()` doesn't allocate, in the second it allocates an object with 1-2 fields but without an array, so `List.of()` is likely to be more memory-saving

Reply:
Even in the no exceptions case, the `exceptionTypes` array still has to be allocated/copied by `Method.getExceptionTypes()`[^1] when the `ProxyMethod` constructor[^2] is invoked.

So if anything, the `List.of(…)` call should be moved into the `ProxyMethod` constructor.

[^1]: https://github.com/openjdk/jdk/blob/e765c42aa71a25a9c30f3409b8fdc6bda6f7b639/src/java.base/share/classes/java/lang/reflect/Method.java#L340-L343
[^2]: https://github.com/openjdk/jdk/blob/e765c42aa71a25a9c30f3409b8fdc6bda6f7b639/src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java#L710-L714

Reply:
> So if anything, the List.of(…) call should be moved into the ProxyMethod constructor. And maybe the call to Method.getExceptionTypes() should be changed to Method.getSharedExceptionTypes()

Makes sense. Do you want me to do this within this PR (this would be a deviation from ticket's target), or it's better to create another one?

Reply:
Ok, I've reverted `ProxyGenerator` and will include the changes into another PR